### PR TITLE
Chat History: Fix 'N months ago'

### PR DESCRIPTION
**Pull request by abeatrix**
CLOSE https://github.com/sourcegraph/cody/issues/3838

This PR refactors the chat history grouping logic where we hard coded "N months ago" for char older than a month.

Replaced current logic with the one provided by @toolmantim :

- This week (7 days ago to now)
- This month (if now > 7 days into the month, start of month to 7 days ago)
- Last month (start of month (this month - 1) to end of month (this month -1)
- More than a month (end of month (this month -1) to infinity)

This provides a more flexible way to group recent chats that will work consistently regardless of the current date.

## Changes

- Switched from using date comparisons to group chats to using the new `getRelativeChatPeriod` utility to group chats into time buckets "Today", "This Week", "This Month", "n months ago" etc.

- Store the grouped chats in a Map rather than separate arrays.

- Reverse the chat history entries before grouping so the most recent chats appear first within each time period group.

- Updated the `groupCodyChats` function to return a Map rather than an object literal.

## Test plan

- Verified chat history is correctly grouped into recent time period buckets.
- Confirmed most recent chats appear first within each time period group.
- Confirmed chats older than last month (if you have any) are sorted into the correct n month group:

image

### Before

From issue:
image

### CODE REVIEW

File: vscode/src/services/HistoryChat.ts [hunk 1]
```diff
@@ -105,7 +105,7 @@ export function groupCodyChats(authStatus: AuthStatus | undefined): GroupedChats
         Yesterday: yesterdayChats.reverse(),
         'This month': thisMonthChats.reverse(),
         'Last month': lastMonthChats.reverse(),
-        'N months ago': nMonthsChats.reverse(),
+        'Before last month': beforeLastMonthChats.reverse(),
```
Comment (abeatrix) [inline vscode/src/services/HistoryChat.ts:108]: Does `Before last month` sound right to you? @toolmantim

File: vscode/src/services/HistoryChat.ts [hunk 2]
```diff
@@ -75,7 +75,7 @@ export function groupCodyChats(authStatus: AuthStatus | undefined): GroupedChats
             const chatTitle = chats[id].chatTitle || getChatPanelTitle(lastHumanText, false)
 
             const lastInteractionTimestamp = new Date(entry.lastInteractionTimestamp)
-            let groupLabel = 'N months ago'
+            let groupLabel = 'Before last month'
```
Comment (philipp-spiess) [inline vscode/src/services/HistoryChat.ts:78]: Nit since we use this string in three places, ca we make it a const in the module scope?

**Review by philipp-spiess — APPROVED** [review]
Code LGTM but I'll let a native speaker weigh in on the naming 🙈

File: vscode/src/services/HistoryChat.ts (hunk 2 above)
Comment (abeatrix) [reply vscode/src/services/HistoryChat.ts]: Yea i've updated the PR to implemented Tim's design with your suggestion too :D

**Pull request merged**